src/reflect/value.go

```diff
@@ -882,8 +974,50 @@ func (v Value) Convert(t Type) Value {
 	panic("unimplemented: (reflect.Value).Convert()")
 }
 
+//go:linkname slicePanic runtime.slicePanic
+func slicePanic()
+
 func MakeSlice(typ Type, len, cap int) Value {
-	panic("unimplemented: reflect.MakeSlice()")
+	if typ.Kind() != Slice {
+		panic("reflect.MakeSlice of non-slice type")
+	}
+
+	rtype := typ.(*rawType)
+
+	ulen := uint(len)
+	ucap := uint(cap)
+	maxSize := (^uintptr(0)) / 2
+	elementSize := rtype.elem().Size()
+	if elementSize > 1 {
+		maxSize /= uintptr(elementSize)
+	}
+	if ulen > ucap || ucap > uint(maxSize) {
+		slicePanic()
+	}
+
+	size, ok := mulNoOverflow(elementSize, uintptr(ucap))
+	if !ok {
+		panic("reflect.MakeSlice: slice size out of range")
+	}
```

Comment:
As far as I can see, this check is now unnecessary because of the check above: `ucap > uint(maxSize)` also checks for overflows.

Reply:
I had that thought too. I'll remove it.